platform/api/brightbox/api.go

```diff
@@ -0,0 +1,234 @@
+func (a *API) DeleteImage(id string) error {
+	if _, err := a.client.DestroyImage(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying image from API: %w", err)
+	}
+
+	return nil
+}
+
+// DeleteCloudIP will remove a cloud IP from Brightbox.
+func (a *API) DeleteCloudIP(id string) error {
+	if _, err := a.client.DestroyCloudIP(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying cloud IP from API: %w", err)
+	}
+
+	return nil
+}
+
+func (a *API) DeleteServer(id string) error {
+	// Let's first unassign the cloud IP.
+	s, err := a.client.Server(context.TODO(), id)
+	if err != nil {
+		return fmt.Errorf("getting server from API: %w", err)
+	}
+
+	var cloudIP string
+	if s != nil && len(s.CloudIPs) >= 1 {
+		cloudIP = s.CloudIPs[0].ID
+	}
+
+	if cloudIP != "" {
+		if _, err := a.client.UnMapCloudIP(context.TODO(), cloudIP); err != nil {
+			return fmt.Errorf("unmaping cloud IP from API: %w", err)
+		}
+		plog.Info("Cloud IP released.")
+	}
+
+	if _, err := a.client.DestroyServer(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying server from API: %w", err)
+	}
+
+	return nil
+}
+
+func (a *API) GC(gracePeriod time.Duration) error {
+	threshold := time.Now().Add(-gracePeriod)
+	// TODO: CloudIP has no creation date for now.
+	// We can't safely delete "old" cloud IPs.
+
+	servers, err := a.client.Servers(context.TODO())
+	if err != nil {
+		return fmt.Errorf("listing servers from API: %w", err)
+	}
+
+	for _, server := range servers {
+		if server.Status == serverstatus.Deleted || server.CreatedAt.After(threshold) {
+			continue
+		}
+
+		if err := a.DeleteServer(server.ID); err != nil {
+			return fmt.Errorf("deleting server: %w", err)
+		}
+	}
+
+	images, err := a.client.Images(context.TODO())
+	if err != nil {
+		return fmt.Errorf("listing servers from API: %w", err)
+	}
+
+	for _, image := range images {
+		if image.Public || image.Status == imagestatus.Deleted || image.CreatedAt.After(threshold) {
+			continue
+		}
+
+		if err := a.DeleteImage(image.ID); err != nil {
+			return fmt.Errorf("deleting image: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (a *API) GetConsoleOutput(id string) (string, error) {
+	// NOTE: There is no way to get console output from the API.
+	// A workaround would be the fetch the console_url + console_token to read from this
+	// endpoint.
+	return "", nil
+}
+
+// UploadImage will uploade an image from the URL on Brightbox and wait for it to become
+// available.
+func (a *API) UploadImage(name, URL string) (string, error) {
+	defaultUsername := "core"
+
+	img, err := a.client.CreateImage(context.TODO(), brightbox.ImageOptions{
+		Name:     &name,
+		URL:      URL,
+		Username: &defaultUsername,
+		Arch:     arch.X86_64,
+	})
+	if err != nil {
+		return "", fmt.Errorf("creating image from API: %w", err)
+	}
+
+	// It usually takes around 20 seconds to extract the image.
+	if err := util.WaitUntilReady(1*time.Minute, 5*time.Second, func() (bool, error) {
```

Comment:
```suggestion
	if err := util.WaitUntilReady(2*time.Minute, 5*time.Second, func() (bool, error) {
```
In case it might take longer it could help to relax the waiting time more